### PR TITLE
feat: show progress bar when cloning a github repo

### PR DESCRIPTION
**Pull request by mooyg**

### CODE REVIEW

**Comment by mooyg** [comment]
Used two libraries [progress bar lib](https://crates.io/crates/progress_bar) and [git2](https://crates.io/crates/git2).

File: src/command.rs [hunk 1]
```diff
@@ -1,13 +1,37 @@
...
+use git2::{build::RepoBuilder, Error, FetchOptions, RemoteCallbacks, Repository};
+use progress_bar::{
+    color::{Color, Style},
+    progress_bar::ProgressBar,
+};
+
+use crate::size::get_repo_size;
+
+pub fn git_clone(proj_name: &str, url: String) -> Result<Repository, Error> {
+    println!("PROJ NAME = {}", &proj_name);
```
Comment (mintbomb27) [inline src/command.rs:13]: Need not print the project again

File: src/command.rs [hunk 2]
```diff
@@ -1,13 +1,37 @@
...
+    let repo_size = get_repo_size(&url);
+
+    let mut progress_bar = ProgressBar::new(repo_size / 1000);
```
Comment (mintbomb27) [inline src/command.rs:16]: this might vary and even make the repo size 0. Why not set a 0 to 100% and map the repo size to that range?

**Review by mintbomb27 — CHANGES_REQUESTED** [review]
Please see the comments :)

**Comment by mooyg** [comment]
Fixed

**Review by mintbomb27 — APPROVED** [review]
LGTM! Thanks for your contribution!

**Pull request merged**